AP-186: Create a file input (#316)
Atom-Learning/components: src/components/file-input/FileInput.mdx

---
title: File Input
component: FileInput
description: FileInput is the lowest-level input component for the file type
category: Form primitives
---

`FileInput` is a specialised input that allows file uploads.

The 3 most important props it takes are: `onSelect`, `id` and `label`.

`onSelect` is a callback that accepts the actual file selection (as a FileList), which gets updated after the user interacts with the file browser interface. The file selection can then be used in order to build a payload for something like a `POST` API call.

> Note: The actual action of uploading the file to a backend is not this input's responsability.

Reviewer: ```suggestion
Note: The actual action of uploading the file to a backend is the parent's responsibility.
```

> `id` is really important because under the hood, the actual file input is hidden, while the visible element is a `label` with an `htmlFor` attribute that pairs with the hidden input via the `id` attribute. So make sure the `id` is unique.

Reviewer: ```suggestion
`id` is really important because the actual file input is hidden, while the visible element is a `label` with an `htmlFor` attribute that pairs with the hidden input via the `id` attribute. 

 Note: `id` has to be unique 
```
This way it makes it stand out that the id has to be unique
